Midterm Demo Integration

creallfluharty opened the pull request.

creallfluharty committed:
stub out login endpoint

creallfluharty committed:
fix get workspace

chwebb02 commented on user/controller.py:
```diff
@@ -59,7 +59,14 @@ def delete():
+    username = request.json['username']
+    password = request.json['password']
+
+    user, token = user_service.login(username, password)
+    return jsonify({
+        "user": json.loads(user.to_json()),  # HACK: same as above, also TODO: if user is null, again depending on AGENDUM-62
+        "token": token,
+    }), 200
```
This should probably be handled in the service.  For now this is sufficient, however.

chwebb02 commented on user/service.py:
```diff
@@ -45,3 +45,13 @@ def delete(user_id: str) -> bool:
+    password_hash = hash_password(password + user.password_salt)
+
+    if password_hash != user.password_hash:
+        return
```
We should probably be throwing an exception here.

chwebb02 commented on workspace/controller.py:
```diff
@@ -9,20 +11,23 @@ def create():
+def get(workspace_id: str = None):
     workspaces = ws_service.get(workspace_id)
 
-    if workspaces is not None:
-        return jsonify({'workspaces': workspaces}), 200
+    # HACK: same deal as in user
+    if isinstance(workspaces, list):
+        workspaces_json = [json.loads(workspace.to_json()) for workspace in workspaces]
     else:
-        return jsonify({"error": "Workspace not found"}), 404
+        workspaces_json = json.loads(workspaces.to_json())
```
As before, probably a service layer change.  Fine for now.

mbowman21 committed:
push up

mbowman21 committed:
got edit task working

mbowman21 committed:
got tags working

creallfluharty committed:
add workspace id to task endpoint

creallfluharty committed:
actually filter by workspace id 😅

creallfluharty committed:
don't require JSON body for task endpoint

creallfluharty committed:
make workspace_id required

in other words - don't allow allow API consumers to requests tasks across all workspaces
note that this also means requests for individual tasks must also be scoped to a workspace (even though task ids are globally unique)

mbowman21 committed:
updated the request for the workspace_id
